test/e2e/artifacts_test.go

```diff
@@ -90,46 +93,64 @@ func (s *ArtifactsSuite) TestArtifactGC() {
 				artifactState{"second-on-completion", "my-bucket-2", true, false},
 			},
 		},
+		// entire Workflow based on a WorkflowTemplate
 		{
 			workflowFile:                 "@testdata/artifactgc/artgc-from-template.yaml",
+			hasGC:                        true,
 			expectedGCPodsOnWFCompletion: 1,
 			expectedArtifacts: []artifactState{
 				artifactState{"on-completion", "my-bucket-2", true, false},
 				artifactState{"on-deletion", "my-bucket-2", false, true},
 			},
 		},
+		// entire Workflow based on a WorkflowTemplate
 		{
 			workflowFile:                 "@testdata/artifactgc/artgc-from-template-2.yaml",
+			hasGC:                        true,
 			expectedGCPodsOnWFCompletion: 1,
 			expectedArtifacts: []artifactState{
 				artifactState{"on-completion", "my-bucket-2", true, false},
 				artifactState{"on-deletion", "my-bucket-2", false, true},
 			},
 		},
+		// Step in Workflow references a WorkflowTemplate's template
 		{
 			workflowFile:                 "@testdata/artifactgc/artgc-step-wf-tmpl.yaml",
+			hasGC:                        true,
 			expectedGCPodsOnWFCompletion: 1,
 			expectedArtifacts: []artifactState{
 				artifactState{"on-completion", "my-bucket-2", true, false},
 				artifactState{"on-deletion", "my-bucket-2", false, true},
 			},
 		},
+		// Step in Workflow references a WorkflowTemplate's template
 		{
 			workflowFile:                 "@testdata/artifactgc/artgc-step-wf-tmpl-2.yaml",
+			hasGC:                        true,
 			expectedGCPodsOnWFCompletion: 1,
 			expectedArtifacts: []artifactState{
 				artifactState{"on-completion", "my-bucket-2", true, false},
 				artifactState{"on-deletion", "my-bucket-2", false, false},
 			},
 		},
+		// entire Workflow based on a WorkflowTemplate which has a Step that references another WorkflowTemplate's template
 		{
 			workflowFile:                 "@testdata/artifactgc/artgc-from-ref-template.yaml",
-			expectedGCPodsOnWFCompletion: 0,
+			hasGC:                        true,
+			expectedGCPodsOnWFCompletion: 1,
 			expectedArtifacts: []artifactState{
-				artifactState{"on-completion", "my-bucket-2", false, true},
+				artifactState{"on-completion", "my-bucket-2", true, false},
```

Comment:
the expectation of this test was wrong previously - the test assumed that in the case of a Workflow defining strategy A and the WorkflowTemplate it's based on defining an artifact with Strategy B, that the net result should be A, but it makes more sense for it to be B (artifact level always overrides workflow level).

Reply:
(the only reason the test was operating the other way before was due to our not successfully accessing the `Status.StoredTemplates`)